### DISCUSSION TITLE
Assist with moving authentication to Azure AD and preserving existing user accounts

### DISCUSSION THREAD

**Discussion by Carike** [datalust/seq-tickets#1726] [Help]
Good day,

We would like to configure our Seq instances to use Azure AD and [preserve our existing user accounts](https://docs.datalust.co/docs/azure-active-directory#:~:text=If%20you%20have%20existing%20user%20accounts%20that%20you%20want%20to%20link%20and%20continue%20using%20with%20AAD%2C%20please%20contact%20Support%20for%20migration%20assistance).

Could you assist us with the process?

We also purchased new licenses, but have not yet upgraded our Seq instances. Not sure whether we should first sort the Azure AD auth or first upgrade. 

Thank you in advance for your assistance.

Kind regards

Carike

**Comment by nblumhardt** [comment]
Hello! Thanks for reaching out.

Updating Seq to the latest version is a good idea before any configuration changes 👍 

Depending on how many accounts you have to migrate, the easiest way to preserve them will be, after changing to the AAD authentication provider, renaming the accounts to use their AAD identity as the username (this generally looks like/will be an email address).

If you're migrating from anything other than username/password authentication, you'll also need to unlink them from the previous identity provider with the button shown below:

![image](https://user-images.githubusercontent.com/342712/209316554-3931ec95-01b9-4a0d-b317-a07bdf5ac8cc.png)

If you run into any difficulties, please drop us a line via our support address, we'll be happy to help out.

Best regards,
Nick
